internal/bgp/bgp.go

```diff
@@ -519,8 +522,19 @@ type tcpmd5sig struct {
 // proper TCP MD5 options when the password is not empty. Works by manupulating
 // the low level FD's, skipping the net.Conn API as it has not hooks to set
 // the neccessary sockopts for TCP MD5.
-func dialMD5(ctx context.Context, addr, password string) (net.Conn, error) {
-	laddr, err := net.ResolveTCPAddr("tcp", "[::]:0")
+func dialMD5(ctx context.Context, addr string, srcAddr string, password string) (net.Conn, error) {
+	// If srcAddr exists on any of the local network interfaces, use it as the
+	// source address of the TCP socket. Otherwise, fall back to the IPv6
+	// unspecified address ("[::]") to let the kernel figure out the source
+	// address.
+	var a string
+	if localAddressExists(srcAddr) {
+		a = srcAddr
+	} else {
+		a = "[::]"
+	}
```

Comment:
do this instead, to simplify?
```suggestion

	a := "[::]"
	if localAddressExists(srcAddr) {
		a = srcAddr
	}
```

Reply:
Thought about this but opted for clear rather than clever here. Anyway, I'm fine with your suggestion, too. Fixed.